client/js/utils/utils.js

```diff
@@ -0,0 +1,32 @@
+var _ = {};
+
+_.each = function(collection, callback){
+  if(collection instanceof Array){
+    for(var i = 0; i < collection.length; i++){
+      callback(collection[i], i, collection);
+    }
+  }else{
+    for(var key in collection){
+      callback(collection[key], key, collection);
+    }
+  }
+};
+
+_.map = function(collection, callback){
+  var result = [];
+  _.each(collection, function(item, key, collection){
+    result.push(callback(item, key, collection));
+  });
+  return result;
+};
+
+_.reduce = function(collection, callback, accumulator){
+  var start = accumulator === undefined ? 1 : 0;
+  accumulator = accumulator === undefined ? collection[0] : accumulator;
```

Comment:
collection[0] will not work for objects. for reference, if you want to write reduce to accept an optional accumulator, it should allow for the accumulator to be initialized to whatever the user wants it to be initialized to - even undefined ( it's tough to imagine a scenario, but the code should not make the assumption that it will never happen ). A better way to check if the accumulator exists is to check arguments.length to see if it was passed in or not.